### DISCUSSION TITLE
Delete a flow

### DISCUSSION THREAD

**Discussion by neumann-nico** [PrefectHQ/prefect#5500] [Q&A] [answered]
Hi there,

Is there a way to delete a flow using Python?
In the Prefect Cloud UI you can simply go to the flow and click on delete.

I am using prefect 1.0.0

```python
from prefect import Flow

flow = Flow(...)
flow_id = flow.register(...)  # flow is registered now e.g. in prefect cloud 
flow.run(...)

# ...

delete_flow(flow_id)  # is something like this possible?
```

**Comment by zanieb** [comment, marked as answer]
You can make a request to the GraphQL API

```
mutation {
  delete_flow(input: {flow_id: "..."}) {
    success
  }
}
```

e.g.

```python
from prefect.client import Client

flow_id = "..."

client = Client()
client.graphql(
    """
    mutation {
    delete_flow(input: {flow_id: "%s"}) {
        success
    }
    }
    """ % flow_id
)
```

**Reply by lucascr91** [reply]
One can also delete all versions of a given flow using `delete_flow_group` instead of `delete_flow`. In this case though, one needs to use the flow_group_id instead of flow_id.

**Reply by mzayer-coveo** [reply]
How can you retrieve the `flow_id` or `flow_group_id`? In my code repository, I only have flow names. (In my infra code, I want to ensure when I remove a flow script from my codebase, it gets deleted on Prefect Cloud)

**Reply by zanieb** [reply]
```python
from prefect.backend import FlowView

flow = FlowView.from_flow_name("your-name")
print(flow.flow_id)
```

**Reply by KayJey-Krayen** [reply]
This does not seem to be working in prefect 2.0 
There is not prefect.backend ..
Is there an alternative for this in prefect 2.0 ?

**Reply by zanieb** [reply]
Hi! We have not yet exposed a method in the `OrionClient` to delete a flow in 2.0. Feel free to contribute the change or open an issue to track it.

You can make requests to to the API though, https://docs.prefect.io/api-ref/rest-api/#/Flows/delete_flow_flows__id__delete

**Comment by neumann-nico** [comment]
@madkinsz thanks that works!
